code/search/SearchForm.php

```diff
@@ -112,7 +112,7 @@ public function getClassesToSearch() {
 	 * @param array $data Request data as an associative array. Should contain at least a key 'Search' with all searched keywords.
 	 * @return SS_List
 	 */
-	public function getResults($pageLength = null, $data = null){
```

Comment:
A problem with this is that it's not going to be backwards compatible; If a user has subclassed SearchForm and overridden this method, upgrading to 3.4 would result in a strict method signature mismatch.

A hack you can use to add new properties is to use func_get_args() to get new parameters, but still add them to the PHPDoc. You will need to make sure you cross-PR this change to the master branch without the hack in place, though, and make sure any PR you make to the 3 branch comments the nature of this feature.